Tpetra: silence warning about shadowed gid var

ibaned opened the pull request:
This fixes a warning about a shadowed variable name.
Personally, I think the need to pass a value to `packValueCount` is really causing more trouble than its worth. I don't see a scenario in which a value-dependent packing scheme would give any benefit, and I suspect not all users of `packValueCount` correctly call it for such a change.

tjfulle commented:
@ibaned, the value-dependent packing is used by Stokhos for runtime sized matrix values.

ibaned commented:
@tjfulle, thanks for letting me know, I didn't know that specialization existed. Still, that doesn't affect this PR as I just changed the name of the temporary. In addition, this code explicitly calls the `GO` specialization of `PackTraits`, so it is worth considering whether the "normal" `packValueCount` function could be overloaded or use default arguments to work when no value is passed in. The line of code I modified just makes up a value of zero, which is not even future proof to the case when `GO` packing becomes value-dependent, its just minimally satisfying this unnecessary requirement.

tjfulle commented:
@ibaned , I suspect you are right on all counts.  Packing variable sized matrix values is a thorn in our side :). In fact, @mhoemmen and I are meeting today to talk about related topics.  I'll bring up your comments

ibaned commented:
@tjfulle much appreciated!

as for this PR, it doesn't affect the design at all, so I think I'll kick off the checkin test script unless someone says otherwise.

tjfulle commented:
I see no problem with the pr

mhoemmen commented on packages/tpetra/core/src/Tpetra_Details_packCrsMatrix_def.hpp:
```diff
@@ -426,9 +426,9 @@ packCrsMatrixRow (const ColumnMap& col_map,
+  const GO gid_example = 0; // packValueCount wants this
   const size_t gids_beg = num_ent_beg + num_ent_len;
-  const size_t gids_len = num_ent * PackTraits<GO, BDT>::packValueCount (gid);
+  const size_t gids_len = num_ent * PackTraits<GO, BDT>::packValueCount (gid_example);
```
Since the input value of `packValueCount` doesn't matter for integers (like `GO`), we could just do this:
```
const size_t gids_len = num_ent * PackTraits<GO, BDT>::packValueCount (GO (0));
```
However, your fix is fine too :)

ibaned commented:
Whoops, looks like this went in twice: once by regular merge and once by checkin script.

ibaned commented:
Git resolved it fine I think
